### DISCUSSION TITLE
Stable-baselines3 in c# with Torchsharp: Gym-based Reinforcement Learning framework

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [xamarin/TorchSharp#1333] [Ideas]
Implementation of **stable-baselines3** in c# with Torchsharp

Someone attempted in the past to port sb3 to c# using Torchsharp. Perhaps this [**rust**-based approach](https://github.com/will-maclean/sb3-burn) could serve as a reference of achieving the goal.

[Stable-baselines3](https://github.com/DLR-RM/stable-baselines3/tree/master) port written in rust using the [burn](https://github.com/tracel-ai/burn) library.

https://github.com/will-maclean/sb3-burn/issues/36

**Comment by asieradzk** [comment]
Hi George. Have you been paying attention to [RLMatrix](https://github.com/asieradzk/RL_Matrix)? Tis about to reach feature parity of Rlib soon. You get shiny PPO and DQN-Rainbow out of the box with distributed reinforcement learning and as much optimization as I was able to squeeze out of C#. For networked stuff I am using SignalR with MessagePack serialisaton.

You're welcome to join the discussion, I am still debating what to use to serialise replay buffers...

I am not finished with networking stuff but I did a quick commit so you can have a [look](https://github.com/asieradzk/RL_Matrix/tree/master/examples/Networked):
